curl/inc/leatherman/curl/client.hpp

```diff
@@ -216,10 +276,25 @@ namespace leatherman { namespace curl {
         LEATHERMAN_CURL_NO_EXPORT void set_ca_info(context& ctx);
         LEATHERMAN_CURL_NO_EXPORT void set_client_protocols(context& ctx);
 
+        LEATHERMAN_CURL_NO_EXPORT void cleanup_temp_file(boost::filesystem::path const& temp_path, FILE* tfp, request const& req, std::string const& file_path, std::string const& reason);
```

Comment:
I'm going to clean it up separately, but this could be done as a forward-declaration rather than including the header file.